### PR TITLE
Rau checkupdate needs permission check otherwise crashes

### PR DESCRIPTION
**Pull request by muhammedyalcin**

### CODE REVIEW

**Comment by alperozisik** [comment]
@muhammedyalcin why phone state is required?

**Comment by muhammedyalcin** [comment]
By new feature of rau report, we need permission of user to read their device id.

**Comment by muhammedyalcin** [comment]
Relating to SUPDEV-1052 this issue we have updated sf-core. According to that updating, location way of implementation is updated & In case of different each other (language  and region), speectotext won't work properly. To eliminate this issue i implemented locale property in case of diff from each other.

File: lib/location.js [hunk 1]
```diff
@@ -69,6 +69,6 @@ function getLocation(callback) {
         pendingLocationRequests.push(function(location) {
             callback(null, location);
         });
-        Location.start(Location.android.Provider.AUTO);
+        Location.start([Location.Android.Provider.GPS,Location.Android.Provider.NETWORK]);// This way of implementation  supports after 3.0.1 sf-core version
```
Comment (alperozisik) [inline lib/location.js]: It is better to support earlier version too.
I am going to approve this, in order to make modifications on it

**Pull request merged**